observability/micrometer/src/main/java/org/wildfly/extension/micrometer/MicrometerSubsystemSchema.java

```diff
@@ -33,9 +35,25 @@ public VersionedNamespace<IntVersion, MicrometerSubsystemSchema> getNamespace()
 
     @Override
     public PersistentResourceXMLDescription getXMLDescription() {
-        return factory(this)
-                .builder(MicrometerSubsystemRegistrar.PATH)
-                .addAttributes(MicrometerSubsystemRegistrar.ATTRIBUTES)
-                .build();
+        PersistentResourceXMLDescription.Factory factory = factory(this);
+        PersistentResourceXMLDescription.Builder builder =
+                factory.builder(MicrometerSubsystemRegistrar.PATH);
+
+        builder.addAttributes(MicrometerSubsystemRegistrar.ATTRIBUTES.stream());
+        if (this.since(VERSION_2_0)) {
+            builder.addChild(factory.builder(OtlpRegistryDefinitionRegistrar.PATH)
+                    .addAttributes(OtlpRegistryDefinitionRegistrar.ATTRIBUTES.stream())
+                    .setXmlElementName("otlp-registry")
+                    .build());
+        } else {
+            builder.addChild(factory.builder(OtlpRegistryDefinitionRegistrar.PATH)
+                    .addAttributes(MicrometerSubsystemRegistrar.ENDPOINT,
+                            MicrometerSubsystemRegistrar.STEP)
+                    .setXmlElementName("otlp-registry")
+                    .build()
```

Comment:
AIUI the if and else clauses do the same thing. Should the 'addChild' statement be removed from the 'else'?

Reply:
Ah, that's a good catch. The issue here is that the model is changing, but the XML structure is not, and it looks like a little noise was left here in trying to figure out how to handle that correctly. The two branches are slightly different in the the attribute defs come from different places, though the defs themselves are identical, so that is probably a "distinction without a difference". :)

Reply:
Ah, well. Not QUITE identical. The definition on `MicrometerSubsystemRegistrar` includes `.addFlag(AttributeAccess.Flag.ALIAS)` which I added, iirc, due to some runtime errors at one point related to attribute translation. That might also be noise. @pferraro ?

Reply:
The set of attributes of the root resource should be different between version 1.0 and 2.0, correct?

Reply:
Probably. I'll dig into that.